src/validation.h

```diff
@@ -181,12 +181,47 @@ void UnlinkPrunedFiles(const std::set<int>& setFilesToPrune);
 /** Prune block files up to a given height */
 void PruneBlockFilesManual(int nManualPruneHeight);
 
-/** (try to) add transaction to memory pool
- * plTxnReplaced will be appended to with all transactions replaced from mempool
- * @param[out] fee_out optional argument to return tx fee to the caller **/
-bool AcceptToMemoryPool(CTxMemPool& pool, TxValidationState &state, const CTransactionRef &tx,
-                        std::list<CTransactionRef>* plTxnReplaced,
-                        bool bypass_limits, bool test_accept=false, CAmount* fee_out=nullptr) EXCLUSIVE_LOCKS_REQUIRED(cs_main);
+/**
+* Validation result for a single transaction mempool acceptance.
+*/
+struct MempoolAcceptResult {
+    /** Used to indicate the results of mempool validation,
+    * including the possibility of unfinished validation.
+    */
+    enum class ResultType : uint8_t {
+        UNFINISHED, //!> Not fully validated.
+        VALID, //!> Fully validated, valid.
+        INVALID, //!> Invalid.
+    };
+    ResultType m_result_type;
+    TxValidationState m_state;
+
+    // The following fields are only present when m_result_type = ResultType::VALID
+    /** Mempool transactions replaced by the tx per BIP 125 rules. */
+    std::optional<std::list<CTransactionRef>> m_replaced_transactions;
+    /** Raw base fees. */
+    std::optional<CAmount> m_base_fees;
+
+    /** Constructor for failure or unfinished case */
+    explicit MempoolAcceptResult(TxValidationState state, bool finished=true) :
+        m_state(state), m_replaced_transactions(nullopt), m_base_fees(nullopt) {
+            m_result_type = finished ? ResultType::INVALID : ResultType::UNFINISHED;
```

Comment:
You can set m_result_type in the initializer list:

```suggestion
        m_result_type(finished ? ResultType::INVALID : ResultType::UNFINISHED),
        m_state(state), m_replaced_transactions(nullopt), m_base_fees(nullopt) {}
```

Reply:
(Will do this 👀 )